pkg/remediation/remediation.go

```diff
@@ -0,0 +1,162 @@
+package remediation
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/rs/zerolog/log"
+)
+
+// Report includes all query results
+type Report struct {
+	Queries []Query `json:"queries"`
+}
+
+// Query includes all the files that presents a result related to the queryID
+type Query struct {
+	Files   []File `json:"files"`
+	QueryID string `json:"query_id"`
+}
+
+// File presents the result information related to the file
+type File struct {
+	FilePath        string `json:"file_name"`
+	Line            int    `json:"line"`
+	Remediation     string `json:"remediation"`
+	RemediationType string `json:"remediation_type"`
+	SimilarityID    string `json:"similarity_id"`
+}
+
+// Remediation presents all the relevant information for the fix
+type Remediation struct {
+	Line         int
+	Remediation  string
+	SimilarityID string
+	QueryID      string
+}
+
+// RemediationSet includes all the replacements and additions related to a file
+type RemediationSet struct {
+	Replacement []Remediation
+	Addition    []Remediation
+}
+
+// RemediateFile remediationSets the replacements first and secondly, the additions sorted down
+func (s *Summary) RemediateFile(filePath string, remediationSet RemediationSet) error {
+	filepath.Clean(filePath)
+	content, err := os.ReadFile(filePath)
+
+	if err != nil {
+		log.Error().Msgf("failed to read file: %s", err)
+		return err
+	}
+
+	lines := strings.Split(string(content), "\n")
+
+	// do replacements first
+	if len(remediationSet.Replacement) > 0 {
+		for i := range remediationSet.Replacement {
+			r := remediationSet.Replacement[i]
+			remediatedLines := replacement(r, lines)
+			if len(remediatedLines) > 0 && willRemediate(remediatedLines, filePath, &r) {
+				lines = s.writeRemediation(remediatedLines, lines, filePath, r.SimilarityID)
+			}
+		}
+	}
+
+	// do additions after
+	if len(remediationSet.Addition) > 0 {
+		// descending order
+		sort.Slice(remediationSet.Addition, func(i, j int) bool {
+			return remediationSet.Addition[i].Line > remediationSet.Addition[j].Line
+		})
+
+		for i := range remediationSet.Addition {
+			a := remediationSet.Addition[i]
+			remediatedLines := addition(a, &lines)
+			if len(remediatedLines) > 0 && willRemediate(remediatedLines, filePath, &a) {
+				lines = s.writeRemediation(remediatedLines, lines, filePath, a.SimilarityID)
+			}
+		}
+	}
```

Comment:
I think there is no need to sort here we can transverse the slice backward allowing for less complexity